lib/route.js

```diff
@@ -163,11 +164,7 @@ function buildRouting (options) {
   }
 
   function hasRoute ({ options }) {
-    return router.find(
-      options.method,
-      options.url || '',
-      options.constraints
-    ) !== null
+    return router.routes.some(route => route.method === options.method && route.path === options.url && equal(route.opts.constraints, options.constraints || {}))
```

Comment:
Does it make a difference if we just passed `options.constraints` right away?

Reply:
Not sure what you mean by that... in equal?
Anyway, I'm reverting this back, see my comment below.